src/Component/Config/Builder/GridBuilder.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Grid\Config\Builder;
+
+use Sylius\Component\Grid\Config\Builder\Action\ActionInterface;
+use Sylius\Component\Grid\Config\Builder\Field\FieldInterface;
+use Sylius\Component\Grid\Config\Builder\Filter\FilterInterface;
+
+final class GridBuilder implements GridBuilderInterface
+{
+    private const DEFAULT_DRIVER_NAME = 'doctrine/orm';
+
+    private string $name;
+    private string $driver;
+    private array $driverConfiguration = [];
+    private array $fields = [];
+    private array $sorting = [];
+    private array $filters = [];
+    private array $actionGroups = [];
+    private array $limits = [];
+
+    private function __construct(string $name, string $resourceClass)
+    {
+        $this->name = $name;
+        $this->driver = self::DEFAULT_DRIVER_NAME;
+        $this->driverConfiguration['class'] = $resourceClass;
+    }
+
+    public static function create(string $name, string $resourceClass): GridBuilderInterface
+    {
+        return new self($name, $resourceClass);
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function setDriver(string $driver): GridBuilderInterface
+    {
+        $this->driver = $driver;
+
+        return $this;
+    }
+
+    public function setRepositoryMethod($method, array $arguments = []): GridBuilderInterface
+    {
+        $this->driverConfiguration['repository'] = [
+            'method' => $method,
+            'arguments' => $arguments,
+        ];
+
+        return $this;
+    }
+
+    public function addField(FieldInterface $field): self
+    {
+        $this->fields[$field->getName()] = $field;
+
+        return $this;
+    }
+
+    public function orderBy(string $name, string $direction = 'asc'): self
+    {
+        $this->sorting = [$name => $direction];
+
+        return $this;
+    }
+
+    public function addOrderBy(string $name, string $direction = 'asc'): self
+    {
+        $this->sorting[$name] = $direction;
+
+        return $this;
+    }
+
+    public function addFilter(FilterInterface $filter): self
+    {
+        $this->filters[$filter->getName()] = $filter;
+
+        return $this;
+    }
+
+    public function addActionGroup(string $name): self
+    {
+        if (!isset($this->actionGroups[$name])) {
+            $this->actionGroups[$name] = ActionGroup::create($name);
+        }
+
+        return $this;
+    }
+
+    public function addAction(ActionInterface $action, string $group): self
+    {
+        if (!isset($this->actionGroups[$group])) {
+            $this->actionGroups[$group] = ActionGroup::create($group);
+        }
```

Comment:
Shouldn't we reuse addActionGroup
```suggestion
        $this-> addActionGroup($group)
```